components/script/dom/subtlecrypto.rs

```diff
@@ -165,7 +166,14 @@ impl SubtleCryptoMethods for SubtleCrypto {
                         if !valid_usage || key_gen_params.name != key_alg {
                             Err(Error::InvalidAccess)
                         } else {
-                            subtle.encrypt_aes_cbc(key_gen_params, &key, &data)
+                            let cx = GlobalScope::get_cx();
+                            rooted!(in(*cx) let mut array_buffer_ptr = ptr::null_mut::<JSObject>());
+                            match subtle.encrypt_aes_cbc(
+                                key_gen_params, &key, &data, cx, array_buffer_ptr.handle_mut()
+                            ) {
+                                Ok(_) => Ok(array_buffer_ptr.get()),
```

Comment:
It would be safer to pass around a handle to the rooted value, rather than the raw object pointer. There's a risk of the value being garbage collected when this occurs, and using a handle would make this a compile error.

The solution to this problem is to move the rooted! above the enclosing match statement so it outlives the handle that we will return from this branch.